### ISSUE TITLE
Deploying only workers

### ISSUE THREAD

**Issue by thiagolocatelli** [applicaai/spring-boot-starter-temporal#3]
This is not an issue, more of a design question:

First of all, really good work here. I ran some tests and I like your library.

Considering I want to deploy more workers, how can I release a jar file with only workers created instead of the workflow initializers? Maybe adding a feature where, using spring profiles or using a property from `spring.temporal` (making default creating workers, and the user can disable by setting `spring.temporal.disable-workers: true`, I can enable or disable the creation of workers. One example: I have an application with rest apis exposed: These apis will validate the the request and initialize the workflow, but I do not want the same application with workers processing them workflow request. I would have another set of docker containers (for example) running to process the workflow requests.

The money transfer example from Temporal kind makes that separation (https://github.com/temporalio/money-transfer-project-template-java) and I would like to have the same approach if I use your library.

Thank you

**Comment by simcha** [comment]
Hi Thiago,

I am very happy that you wrote to me about multi/no workers issue. The truth is I do not understand it from the beginning and I am afraid I made it wrong in our software. 

I understand you will like to have one jar with workflow interface only no workflow implementation class. Than you can use `makeStub(Class<T> workflowInterface, String workflowName)` from the same factory and as no code will be present no worker will be initialized. Only you need to keep implementation classes out of the jar.

Is it O.K. or not?

Thanks,
Jan

**Comment by thiagolocatelli** [comment]
I think you library is just right and I would not change anything about it. Per my understanding is that you create workflow clients and workers.

What I would do is to create a property to create: workflow clients only, workflow workers only or both (what you are currently doing - which is completely ok). I was thinking a little more after I created the "issue" and my suggestion would be `spring.temporal.deployment-type`, with the following values: `client`, `worker`, `both` with both being default.

`client`: it would create workflow clients only, no worker (no worker would be created but workflows could be initialized using workflow clients)
`worker`: it would create workflow workers only, no client (no workflow would be created when deployed, only processed)
`both`: its the behavior you currently have, the same jar file would be able to initialize and process workflows.

I would not change your annotations, I would look into the class `WorkflowAnnotationBeanPostProcessor.java` and use the properties to create or not workers based on the property. I need to look at the code now and see how to disable clients if the property is 'worker'.

does that make sense?

**Comment by simcha** [comment]
Starting from the bottom up:
You can have many workflows.
You right with the @Component.
As for the deployment type we do not create clients automatically so it will not make sense, I think.

**Comment by simcha** [comment]
But some sort of setting is easy to be added and I can do it tomorrow or by the end of the week, if my work will allow. In the meantime there is a spring-profile "temporal_test" that works exactly as you would like, probably. So you can use it for now and tel me is it exactly what you were thinking of. The name is for sure wrong as in future it can change more settings but also I'm not sure if we should use the profile mechanism at all for the setting you think about.

**Comment by thiagolocatelli** [comment]
So the clients are create only when we try to initialized when we programatically create an workflow to initiate it, correct?

**Comment by thiagolocatelli** [comment]
profile may not be a good idea for my suggestion, I would say a property to disable/enable workers would work best.

**Comment by simcha** [comment]
Yes clients (stubs) are one time only so they need to be created every time.

**Comment by the-fine** [comment]
Maybe another approach to this can be an annotation on an Activity implementation, and the `registerActivitiesImplementations` will be called on these implementations?
This way, you can have one service with only the workflow and activity stubs, and one service with the activity implementations, each can be scaled independently.
Does that makes sense @simcha?

**Comment by thiagolocatelli** [comment]
you still need the workflow implementation for the worker, not just the activity implementation, right?

**Comment by the-fine** [comment]
No, i want two separate services, one with only the workflow implementation and one with activity implementation

**Comment by thiagolocatelli** [comment]
My understanding is that you can't have a worker without the worklow implementation type, but I am may be wrong.

```
worker.registerWorkflowImplementationTypes(workflowFactory.makeWorkflowClass(targetClass));
```

**Comment by the-fine** [comment]
there is an example for that here https://docs.temporal.io/docs/java-quick-start#activities (`GettingStartedActivityWorker`)

**Comment by thiagolocatelli** [comment]
How about this example where the worker uses the worflow and activity implementations?

https://github.com/temporalio/money-transfer-project-template-java/blob/master/src/main/java/moneytransferapp/MoneyTransferWorker.java

So if you want to deploy the spring boot in a `worker only` mode, you still need both implementations to work. I may be looking at it the wrong way.

**Comment by the-fine** [comment]
In this example you register both the workflow and the activity to the worker, but its not mandatory to have both registered, it can be either workflow or activity as you can see in the example here https://docs.temporal.io/docs/java-quick-start#activities

**Comment by thiagolocatelli** [comment]
Understood. I will try that example with only activities and see if it works. Thank you

**Comment by the-fine** [comment]
I hope it does :D as its part of the official documentation.
It won't work with this starter as WorkflowAnnotationBeanPostProcessor creates the worker only if there is a TemporalWorkflow annotated class

**Comment by thiagolocatelli** [comment]
Agreed. So for my suggestion I think a property to disable worker creation would work. This way I can disable the workers and have my application deployed with only clients sending workflow requests. Then I can deploy the same application in another cluster having the workers only processing the workflow requests.

**Comment by the-fine** [comment]
@thiagolocatelli sorry for diverting from the main topic of this issue, ill create a new one about this

**Comment by simcha** [comment]
Fixed in 0.5.1

**Comment by thiagolocatelli** [comment]
@simcha I saw that you made the change but the maven dependency is not available on maven central. Am I missing anything?

`Could not find artifact com.github.applicaai:spring-boot-starter-temporal:jar:0.5.1-SNAPSHOT`
